src/main/java/Aibles_Training/java/Spring/boot/active/account/service/impl/AuthServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package Aibles_Training.java.Spring.boot.active.account.service.impl;
+
+import Aibles_Training.java.Spring.boot.active.account.dto.request.OtpRequest;
+import Aibles_Training.java.Spring.boot.active.account.dto.response.OtpResponse;
+import Aibles_Training.java.Spring.boot.active.account.entity.Account;
+import Aibles_Training.java.Spring.boot.active.account.exception.InvalidRequestException;
+import Aibles_Training.java.Spring.boot.active.account.exception.NotFoundException;
+import Aibles_Training.java.Spring.boot.active.account.exception.InvalidOtpException;
+import Aibles_Training.java.Spring.boot.active.account.repository.AuthRepository;
+import Aibles_Training.java.Spring.boot.active.account.service.AuthService;
+import Aibles_Training.java.Spring.boot.active.account.service.RedisService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+@Slf4j
+public class AuthServiceImpl implements AuthService {
+    private final AuthRepository authRepository;
+    private final RedisService redisService;
+    public AuthServiceImpl(AuthRepository authRepository, RedisService redisService) {
+        this.authRepository = authRepository;
+        this.redisService = redisService;
+    }
+
+    @Override
+    public OtpResponse activeAccount(OtpRequest request) {
+        String username = request.getUsername();
+        String otp = request.getOtp();
+        validateInput(username, otp);
+        log.info("Looking for user account with username: {}", username);
+        Optional<Account> otpAccount = authRepository.findUserAccountByUsername(username);
+        if (otpAccount.isEmpty()) {
+            log.error("Username was not registered", username);
+            Map<String, String> errorDetails = new HashMap<>();
+            errorDetails.put("message", "User not found");
+            throw new NotFoundException(errorDetails);
+        }
+
+        Account account = otpAccount.get();
+        if (account.isActivated()) {
+            return new OtpResponse("success", System.currentTimeMillis(), null);
```

Comment:
Use base response for Otp response. 
Service layer must not know what is the response that will be returned to client